### ISSUE TITLE
EL7 :: 1.22.[2-3] :: AttributeError: module 're' has no attribute 'Match'

### ISSUE THREAD

**Issue by adriansev** [kellyjonbrazil/jc#337]
on EL7 (python 3.6.8), starting with 1.22.2 (1.22.1 is OK), i see in command line this error:
```
[root@fst04 ~]# jc --help
jc:  Error - Exit due to unexpected error:
             AttributeError: module 're' has no attribute 'Match'
```

Any idea on this?
Thanks a lot!

**Comment by kellyjonbrazil** [comment]
Hi @adriansev! Thanks for reporting this. I can no longer automatically test python 3.6 (only 3.7 and above) so these things can slip through on older versions. 😃 

I'll take a look at the diff between 1.22.1 and 1.22.2, but could you run the following, which might help narrow it down?
```
jc --help -dd
```
Thanks!

**Comment by adriansev** [comment]
@kellyjonbrazil first, let me just say "Thank *you*" for this great tool :)
So, the output is this https://pastebin.com/mT5u3vRS
Let me know what else i can do :)
Thanks a lot!

**Comment by kellyjonbrazil** [comment]
Ok, I see what happened there. I've been adding type annotations and it looks like I used the `re.Match` class as a type in the new `findmnt` parser.

A quick fix might be to copy/paste the `findmnt` parser file to your plugin directory.
https://github.com/kellyjonbrazil/jc/blob/master/jc/parsers/findmnt.py

and change line 143 from:
```
def _replace(matchobj: re.Match) -> str:
```
to
```
def _replace(matchobj):
```
That is, unless there are other parsers I got fancy with. Looks like python 3.6 doesn't support that particular class like the newer versions.

Here are the parser plugin directories (by OS):
- Linux/unix: `$HOME/.local/share/jc/jcparsers`
- macOS: `$HOME/Library/Application Support/jc/jcparsers`
- Windows: `$LOCALAPPDATA\jc\jc\jcparsers`

**Comment by adriansev** [comment]
@kellyjonbrazil yeah, that worked, but i had to also modify `parsers/ifconfig.py:329`
i added Any to imports from typing and replace `re.Match` usage with Any
Thanks a lot!

**Comment by kellyjonbrazil** [comment]
I'll have to figure out what to do here, long term. Python 3.6 is not officially supported, so I don't have automated tests for it, so things like this may slip through again in the future. I can relax the typing for these two parsers in the next release and keep that in mind for other parsers using regular expression typing.

**Comment by adriansev** [comment]
@kellyjonbrazil well, personally (and lots of admins in scientific community) would love if you could support 3.6 for the duration of EL7 life (with my admin hat, the "support" of python is given by distro lifetime and usage :D ;) ) but if it's inconvenient (for lot's of reasons, i understand this very well :) ), you could cut a last tag with 3.6 support and after that publish only >=3.9 support.

**Comment by kellyjonbrazil** [comment]
If I'm interpreting that correctly, it looks like Extended life cycle support (ELS) add-on for EL7 ends on July 30th, 2026.

https://access.redhat.com/support/policy/updates/errata

I'll see if I can get python 3.6 back into automated tests in Github Actions.

**Comment by kellyjonbrazil** [comment]
It looks like they may have added python 3.6 back into the latest Ubuntu packages available via Github Actions, so I may be able to add automated 3.6 tests again. I'll look into this once I start working on the next release.

**Comment by adriansev** [comment]
thanks a lot!!! should i close this issue as the problem is both acknowledged and understood?

**Comment by kellyjonbrazil** [comment]
Tests are now passing in Github Actions for python 3.6. Using `ubuntu-20.04` instead of `ubuntu-latest`, which includes python 3.6 - 3.11. No change to `macos-latest` and `windows-latest` instances.

**Comment by kellyjonbrazil** [comment]
Fixed in v1.22.4.
